go/cmd/dolt/commands/sql.go

```diff
@@ -261,30 +262,54 @@ func (cmd SqlCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 		if isTty {
 			err := execShell(sqlCtx, queryist, format)
 			if err != nil {
-				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+				return sqlHandleVErrAndExitCode(queryist, errhand.VerboseErrorFromError(err), usage)
 			}
 		} else if runInBatchMode {
 			se, ok := queryist.(*engine.SqlEngine)
 			if !ok {
 				misuse := fmt.Errorf("Using batch with non-local access pattern. Stop server if it is running")
-				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(misuse), usage)
+				return sqlHandleVErrAndExitCode(queryist, errhand.VerboseErrorFromError(misuse), usage)
 			}
 
 			verr := execBatch(sqlCtx, se, input, continueOnError, format)
 			if verr != nil {
-				return HandleVErrAndExitCode(verr, usage)
+				return sqlHandleVErrAndExitCode(queryist, verr, usage)
 			}
 		} else {
 			err := execMultiStatements(sqlCtx, queryist, input, continueOnError, format)
 			if err != nil {
-				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+				return sqlHandleVErrAndExitCode(queryist, errhand.VerboseErrorFromError(err), usage)
 			}
 		}
 	}
 
 	return 0
 }
 
+// sqlHandleVErrAndExitCode is a helper function to print errors to the user. Currently, the Queryist interface is used to
+// determine if this is a local or remote execution. This is hacky, and too simplistic. We should possibly add an error
+// messaging interface to the CliContext.
+func sqlHandleVErrAndExitCode(queryist cli.Queryist, verr errhand.VerboseError, usage cli.UsagePrinter) int {
+	if verr != nil {
+		if msg := verr.Verbose(); strings.TrimSpace(msg) != "" {
+			if _, ok := queryist.(*engine.SqlEngine); !ok {
+				// We are in a context where we are attempting to connect to a remote database. These errors
+				// are unstructured, so we add some additional context around them.
+				msg = fmt.Sprintf("Error connecting to remote database: \"%s\".", msg)
```

Comment:
Is this only for the case when we are connecting to a local database via a TCP connection? Just wondering if we can avoid saying "remote" and be more specific in this error message. I think there's a good chance that customers trying to use `dolt sql` will get confused by this error message and need to ask for support since they aren't trying to connect off box.

Reply:
I updated the message to acknowledge that this is undocumented, and to reach out to me for help:

```
You've encountered a new behavior in dolt which is not fully documented yet.
A local dolt server is using your dolt data directory, and in an attempt to service your request, we are attempting to
connect to it. That has failed. You should stop the server, or reach out to @macneale on https://discord.gg/gqr7K4VNKe
for help.
A local server is running, and dolt is failing to connect. Error connecting to remote database: "Error 1045: User not found 'root'".
```